### ISSUE TITLE
Failed to minify use create-react-app “npm run build”

### ISSUE THREAD

**Issue by wk633** [pubnub/chat-engine#332]
Hi, I am trying to build a chatroom demo using chat-engine. I used **create-react-app** to build this project and have successfully achieved my goal. But when I try to minify my project to get a product level code, it failed.
![](https://ws1.sinaimg.cn/large/006tNc79gy1fp8tytood3j30wy0ngjsc.jpg)

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
2 info using npm@5.6.0
3 info using node@v9.2.1
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle pubnub_demo@0.1.0~prebuild: pubnub_demo@0.1.0
6 info lifecycle pubnub_demo@0.1.0~build: pubnub_demo@0.1.0
7 verbose lifecycle pubnub_demo@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle pubnub_demo@0.1.0~build: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/wk/githome/pubnub_demo/node_modules/.bin:/usr/local/sbin:/Users/wk/Downloads/mongodb-src-r3.4.5/bin:/Users/wk/Library/Python/3.4/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS
9 verbose lifecycle pubnub_demo@0.1.0~build: CWD: /Users/wk/githome/pubnub_demo
10 silly lifecycle pubnub_demo@0.1.0~build: Args: [ '-c', 'node scripts/build.js' ]
11 silly lifecycle pubnub_demo@0.1.0~build: Returned: code: 1  signal: null
12 info lifecycle pubnub_demo@0.1.0~build: Failed to exec build script
13 verbose stack Error: pubnub_demo@0.1.0 build: `node scripts/build.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:285:16)
13 verbose stack     at EventEmitter.emit (events.js:159:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:159:13)
13 verbose stack     at maybeClose (internal/child_process.js:943:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:220:5)
14 verbose pkgid pubnub_demo@0.1.0
15 verbose cwd /Users/wk/githome/pubnub_demo
16 verbose Darwin 17.3.0
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
18 verbose node v9.2.1
19 verbose npm  v5.6.0
20 error code ELIFECYCLE
```

Any quick suggestion to solve it?

**Comment by wk633** [comment]
After use your fork, the problem is resolved. Thanks a lot. 
By the way, how you do this translation? It's seems a lot of work if trying to do it manually(over 1000 lines).

**Comment by AdamTyler** [comment]
I pulled in babel to the environment and let webpack do all the work for me

**Comment by wk633** [comment]
Thank you for your replay.

**Comment by AdamTyler** [comment]
@madisonsmith this is the issue I addressed that got us going. There original issue was here #275 and I think @ianjennings is still incorporating it in #326

**Comment by madisonsmith** [comment]
👍 I'll reopen @wk633 as we plan to have this solved shortly. Thanks for the report!

**Comment by AdamTyler** [comment]
This was fixed in #340
